### PR TITLE
ovirt-datacenters: Add option to cleanup dc which contains hosted engine VM

### PR DESCRIPTION
**Pull request by KKoukiou**
In this case of HE all entities that the hosted_engine VM relies on within the datacenter
will not be deleted.

Also fixed some deprecation warnings.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for an oVirt project! 

I am the bot who triggers CI builds on the oVirt Jenkins server.
As a security measure I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organisation's projects.

Note: You might see multiple messages like this showing up on a single PR, this is due to a known bug in the GitHub trigger plugin on Jenkins, the oVirt infra team is working on fixing this via new testing flow in the near future, so we'll be able to post a single message per PR.

**Comment by KKoukiou** [comment]
@machacekondra Are you ok with this addition to cover HE environments?
@tareqalayan @kobihk what do you think?
I need to do some extra verification but I 'd like to know if you have any objections before I proceed.

File: roles/ovirt-datacenter-cleanup/README.md
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/README.md]: Should we automatically detect the storage of `hosted_engine` or are there any issues with that?

File: roles/ovirt-datacenter-cleanup/README.md
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/README.md]: s/we/the

Comment (machacekondra) [reply roles/ovirt-datacenter-cleanup/README.md]: s/OVirt/oVirt

File: roles/ovirt-datacenter-cleanup/README.md
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/README.md]: s/hosted_engine/<i>hosted_engine</i>

File: roles/ovirt-datacenter-cleanup/defaults/main.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/defaults/main.yml]: Is `false` reasonable default?

File: roles/ovirt-datacenter-cleanup/README.md
Comment (tareqalayan) [inline roles/ovirt-datacenter-cleanup/README.md]: Specify whether the*

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_vm_cluster.yml
Comment (tareqalayan) [inline roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_vm_cluster.yml]: "{{ something }}"
be consistent with whitespaces.

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_vm_cluster.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_vm_cluster.yml]: There is nice filter called `selectattr`[1] maybe you can find it usefull in this use case.

Something like this could work:
```yaml
"{{ ovirt_vms  | selectattr("name", "equalto", hosted_engine_vm_name) }}
```

[1] http://jinja.pocoo.org/docs/dev/templates/#selectattr

Python jina 2.7 should be available on el7, so it shouldn't cause any issues.

File: roles/ovirt-datacenters/README.md
Comment (tareqalayan) [inline roles/ovirt-datacenters/README.md]: oVirt

**Review by tareqalayan — COMMENTED** [review]
some typos

File: roles/ovirt-datacenter-cleanup/tasks/vms.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/tasks/vms.yml:12]: Hmm, this works? Shouldn't it be:

```yaml
when:
  - not hosted_engine_remove or (hosted_engine_remove and ovirt_item.name != hosted_engine_vm_name)
```

File: roles/ovirt-datacenter-cleanup/README.md
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/README.md]: Maybe `hosted_engine_remove` would be more reasonable name?

Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/README.md]: Hmm. I find hosted_engine_remove naming more confusing.
If hosted_engine_remove == true I would expect that the role will actually remove the hosted_engine VM.

Thought hosted_engine name for parameter just implies that the datacenter we are handling contains hosted_engine VM. This VM will not we removed of course.

File: roles/ovirt-datacenter-cleanup/README.md
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/README.md]: I am not aware how to do it. I could not get such info with engine-config.
I can get however the hosted_engine_vm name with engine-config. But I 'll need some regexp there to extract it and i thought it would look ugly.
What do you think?

File: roles/ovirt-datacenter-cleanup/defaults/main.yml
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/defaults/main.yml]: Not sure is hosted_engine installations are more frequent. I sure can change this one, no problem.

File: roles/ovirt-datacenter-cleanup/tasks/vms.yml
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/tasks/vms.yml:12]: Works fine, and it's in docs as well.
check here second example http://docs.ansible.com/ansible/latest/playbooks_conditionals.html#the-when-statement

File: roles/ovirt-datacenter-cleanup/README.md
Comment (kobihk) [inline roles/ovirt-datacenter-cleanup/README.md:11]: Please add the master storage domain which you have to skip the deletion of it too

File: roles/ovirt-datacenter-cleanup/README.md
Comment (kobihk) [inline roles/ovirt-datacenter-cleanup/README.md]: because we know that hosted engine vm name(HostedEngine) and hosted engine storage domain name(hosted_storage) are constants.
maybe it will be better to detect it dynamically...
and remove this  hosted_engine parameter

File: roles/ovirt-datacenter-cleanup/defaults/main.yml
Comment (kobihk) [inline roles/ovirt-datacenter-cleanup/defaults/main.yml]: maybe we can remove it and set it dynamically ... 
 as I explained in roles/ovirt-datacenter-cleanup/README.md

File: roles/ovirt-datacenter-cleanup/tasks/main.yml
Comment (kobihk) [inline roles/ovirt-datacenter-cleanup/tasks/main.yml]: when you have only one condition, IMHO it will be nice to do it in the same line like:
when: not hosted_engine

http://docs.ansible.com/ansible/latest/playbooks_conditionals.html#the-when-statement
they use it in the example as you did just in case of Multiple conditions that all need to be true (a logical ‘and’) can also be specified as a list

File: roles/ovirt-datacenters/README.md
Comment (kobihk) [inline roles/ovirt-datacenters/README.md]: same as my comment in:
roles/ovirt-datacenter-cleanup/README.md

File: roles/ovirt-datacenters/README.md
Comment (kobihk) [inline roles/ovirt-datacenters/README.md]: same as my comment in:
roles/ovirt-datacenter-cleanup/README.md

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: `selectattr('name', 'equalto', hosted_engine_vm_name)` isn't really needed, it's already done by previous task.
I think all you need is just "{{ ovirt_vms | length > 0 }}"

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: I think you don't need this task, because it's not reliable, you should change the task below to:

```yaml
- name: Fetch whole dict for disk of HE VM
      ovirt_disks_facts:
        auth: "{{ ovirt_auth }}"
        pattern: "datacenter={{ data_center_name }} and  vm_names=cirros14 and bootable=true
```

Should you backup all disks of the HE VM or just the bootable one?

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (machacekondra) [inline roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: Maybe just:

```
ovirt_vms[0].cluster.id[0]
```

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: Done

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: search for bootable disk does not work but I think we don't need it anyway, I don't see use case for secondary disk for HE.

File: roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml
Comment (KKoukiou) [reply roles/ovirt-datacenter-cleanup/tasks/find_hosted_engine_details.yml]: @machacekondra Thanks for review!

File: roles/ovirt-datacenters/README.md
Comment (kobihk) [inline roles/ovirt-datacenters/README.md:7]: details

**Review by kobihk — APPROVED** [review]

**Comment by KKoukiou** [comment]
I 'll have to repost, please don't merge now.

**Comment by KKoukiou** [comment]
I found again some problem so since it's still open I 'll repost here.

**Review by KKoukiou — COMMENTED** [review]
https://github.com/oVirt/ovirt-ansible/blob/30add61c3ea89d734d2d4e32c06a3e4134e6883a/roles/ovirt-datacenter-cleanup/tasks/disks.yml#L1-L4

doesn't filter as I expected. I will fix it.

File: roles/ovirt-datacenter-cleanup/README.md
Comment (kobihk) [inline roles/ovirt-datacenter-cleanup/README.md:13]: accommodate

**Comment by KKoukiou** [comment]
Sorry for delay on this, we have some issues with HE deployment and can't proceed yet with patch.

**Comment by machacekondra** [comment]
@KKoukiou Can you please move this patch to: https://github.com/ovirt/ovirt-ansible-infra/ ?

**Comment by KKoukiou** [comment]
@machacekondra Sure, feel free to close this PR and I 'll recreate the patch in the  https://github.com/ovirt/ovirt-ansible-infra/  repo when possible.

**Pull request closed without merging**
